src/bp/nlu-core/entities/microsoft-extractor/index.ts

```diff
@@ -0,0 +1,380 @@
+        return Recognizers.Culture.Spanish
+      default:
+        throw new Error('This language is not supported by Microsoft Recognizer')
+    }
+  }
+
+  private _globalRecognizers = [
+    Recognizers.recognizePhoneNumber,
+    Recognizers.recognizeIpAddress,
+    Recognizers.recognizeMention,
+    Recognizers.recognizeHashtag,
+    Recognizers.recognizeEmail,
+    Recognizers.recognizeURL,
+    Recognizers.recognizeGUID
+  ]
+
+  private _languageDependantRecognizers = [
+    Recognizers.recognizeNumber,
+    Recognizers.recognizeOrdinal,
+    Recognizers.recognizePercentage,
+    Recognizers.recognizeAge,
+    Recognizers.recognizeCurrency,
+    Recognizers.recognizeDimension,
+    Recognizers.recognizeTemperature,
+    Recognizers.recognizeDateTime,
+    Recognizers.recognizeBoolean
+  ]
+
+  public static get entityTypes(): string[] {
+    return []
+  }
+
+  public async configure() {
+    await this._cache.restoreCache()
+  }
+
+  public async extractMultiple(
+    inputs: string[],
+    lang: string,
+    useCache?: boolean
+  ): Promise<EntityExtractionResult[][]> {
+    let options: MicrosoftParams
+
+    if (!SupportedLangsList.includes(lang)) {
+      lang = 'en'
+      options = {
+        lang: 'en' as SupportedLangs,
+        recognizers: [this._globalRecognizers]
+      }
+    }
+
+    options = {
+      lang: lang as SupportedLangs,
+      recognizers: [...this._languageDependantRecognizers, ...this._globalRecognizers]
+    }
+
+    const [cached, toFetch] = this._cache.getCachedAndToFetch(inputs)
+
+    const chunks = _.chunk(toFetch, BATCH_SIZE)
+    const batchedRes = await Promise.mapSeries(chunks, c => this._extractBatch(c, options))
+
+    return _.chain(batchedRes)
+      .flatten()
+      .concat(cached)
+      .orderBy('idx')
+      .map(x => x.entities!)
+      .value()
+  }
+
+  public async extract(input: string, lang: string, useCache?: boolean): Promise<EntityExtractionResult[]> {
+    return (await this.extractMultiple([input], lang, useCache))[0]
+  }
+
+  private formatEntity(entity) {
+    let unit: string
+    let value: string
+    if (entity.resolution.values) {
+      value = entity.resolution.values[0]
+      unit = entity.resolution.values[0].type
+    } else {
+      unit = entity.resolution.unit ? entity.resolution.unit : entity.typeName
+      value = entity.resolution.value
+    }
+
+    const formated: EntityExtractionResult = {
+      confidence: 1.0,
+      type: entity.typeName,
+      value,
+      start: entity.start,
+      end: entity.end + 1,
+      metadata: {
+        source: entity.text,
+        entityId: `system.${entity.typeName}`,
+        extractor: 'system',
+        unit
+      }
+    }
+    return formated
+  }
+
+  private ducklingTypeMappings = {
+    currency: 'amountOfMoney',
+    email: 'email',
+    number: 'number',
+    ordinal: 'ordinal',
+    phonenumber: 'phoneNumber',
+    temperature: 'temperature',
+    url: 'url'
+  }
+
+  private ducklingUnitMapping = {
```

Comment:
feels weird to see such big hardcoded datastructures in a private property...

This should probably go in a separate file with constants (and cap cases) to prevent from obfuscing the core logic of you extractor.